lib/components/map/gtfs-rt-vehicle-overlay.js

```diff
@@ -0,0 +1,218 @@
+import { transit_realtime as transitRealtime } from 'gtfs-realtime-bindings'
+import L from 'leaflet'
+import TransitVehicleOverlay from '@opentripplanner/transit-vehicle-overlay'
+import PropTypes from 'prop-types'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+import { FeatureGroup, MapLayer, Marker, Polyline } from 'react-leaflet'
+import styled from 'styled-components'
+
+export const VehicleShape = styled.span`
+  background-color: #${props => props.route?.route_color || '000000'};
+  border: 1px solid #000000;
+  border-radius: 50%;
+  color: #${props => props.route?.route_text_color || 'ffffff'};
+  display: block;
+  height: 24px;
+  left: -6px;
+  line-height: 24px;
+  margin-left: auto;
+  margin-right: auto;
+  position: relative;
+  text-align: center;
+  top: -6px;
+  width: 24px;
+`
+
+/**
+ * Convert GTFS-rt entities to vehicle location from OTP-UI.
+ * @param vehicle The GTFS-rt vehicle position entity to convert.
+ * @param routes Optional GTFS-like routes list.
+ */
+function gtfsRtToTransitVehicle (gtfsRtVehicle, routes) {
+  const { position, stopId, timestamp, trip, vehicle } = gtfsRtVehicle.vehicle
+  let routeShortName
+  if (routes.length !== 0) {
+    routeShortName = routes.find(r => r.route_id === trip.routeId)?.route_short_name
+  }
+  return {
+    heading: position.bearing,
+    id: vehicle.id,
```

Comment:
Is this id field needed? https://github.com/opentripplanner/otp-ui/blob/e7340f184d047d322b5e2eb995ba8a3a95230ffc/packages/core-utils/src/types.js#L86

Reply:
Thanks for spotting that, it is required and needs to be added to OTP-UI types involved with overlays that use `ZoomBasedMarkers` (you get a key prop required message otherwise). Noting this as an issue against OTP-UI.

Reply:
Oh my goodness. Thanks for opening the issue!